db.go

```diff
@@ -735,11 +735,16 @@ func (m *DbMap) SelectOne(holder interface{}, query string, args ...interface{})
 
 // Begin starts a gorp Transaction
 func (m *DbMap) Begin() (*Transaction, error) {
+	return m.BeginOption(nil)
+}
+
+// BeginOption starts a gorp Transaction with sql.TxOptions
+func (m *DbMap) BeginOption(opt *sql.TxOptions) (*Transaction, error) {
```

Comment:
I think this is a good place for Dave Cheney's functional options (https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis).  We unfortunately can't change the signature of `Begin` without breaking anyone who's using an `interface` type to represent our `DbMap` type - but as long as we're adding a new signature, it may make sense to support a variadic option func argument so that we can add other options without breaking backward compatibility in the future.

```go
type optCtx struct {
    opts *sql.TxOptions
}

type BeginOpt func(optCtx) optCtx

func TxOptoins(opts *sql.TxOptions) BeginOpt {
    return func(ctx optCtx) optCtx {
        ctx.opts = opts
        return ctx
    }
}

func (m *DbMap) BeginOpt(options ...BeginOpt) (*Transaction, error) {
    var opt optCtx
    for _, o := range options {
        opt = o(opt)
    }
    // ... logger stuff ...
    tx, err := begin(m, opt.opts)
    // etc
}
```

Reply:
Do you think keeping optCtx unexported is the way to go? i.e. only gorp can create new options, rather than callers? I don't have a strong argument either way.

Reply:
In the example above, I don't believe that exporting the type helps anyone.  There are no exported functions that accept an option context (other than the BeginOpt functions themselves), so letting callers construct the type doesn't really do anything - especially if the `opts` field is left unexported.

My main argument: it's easy export something that was previously unexported (no one can use it, so changing it *shouldn't* break backward compatibility); it's very hard to unexport something that was previously exported (we'd have to cut a new major version).

So generally, I'm of the opinion that we should start with it unexported; if interest is shown in exporting it, we can always export it then.